docs/source/specs/typespec/main.tsp

```diff
@@ -0,0 +1,251 @@
+import "@typespec/http";
+import "@typespec/rest";
+import "@typespec/openapi3";
+import "@typespec/versioning";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+
+@service({title:"Role Based Access Control for Kessel Project"})
+
+@server("https://localhost:{port}/{basePath}", "Development Server", { basePath: string="api/rbac/v2", port: string="8000" })
+@server("https://console.stage.redhat.com/{basePath}", "Stage Server", { basePath: string="api/rbac/v2" })
+@server("https://console.redhat.com/{basePath}", "Production Server", { basePath: string="api/rbac/v2" })
+@versioned(Versions)
+
+namespace KesselRBAC;
+
+enum Versions{
+    v2,
+}
+
+model Link {
+    @format("uri")
+    first: string = "/api/v2/(resources)/?offset=0&limit=10";
+    @format("uri")
+    previous: string = "/api/v2/(resources)/?offset=20&limit=10";
+    @format("uri")
+    next: string = "/api/v2/(resources)/?offset=40&limit=10";
+    @format("uri")
+    last: string = "/api/v2/(resources)/?offset=90&limit=10";
+}
+
+model Meta {
+    count: int64 = 30
+}
+
+model ListResponse<T> {
+    data: T[];
+    links: Link;
+    meta: Meta;
+}
```

Comment:
I'm not sure about this ListResponse schema for a couple of reasons:

- it relies on offset pagination rather than [keyset/cursor pagination](https://use-the-index-luke.com/no-offset) (Django Rest has [this](https://www.django-rest-framework.org/api-guide/pagination/#cursorpagination) to say about it.) I'm not sure this matters really, though because...
- I think the shape of this API might not warrant pagination regardless, since it is sort of "built-in" paginated to the children of one parent of the hierarchy at a time. Replace "parent" with "page" and it's similar considering limits to the total number of workspaces and limits to the number of children per node. Those limits effectively define the page size.

This may change considering the query problem, so another option which might be pragmatic for now is to simply omit the List endpoint since it is still an active research topic.

Reply:
We may want to get input from platex on this and see if there are other standards/patterns in other platform APIs (I don't believe there is, but it might be worth guidance). Right now the pattern for RBAC in v1 is limit/offset pagination, which would primarily have implications on the UI clients if we moved away from that. Not suggesting we shouldn't, but a consideration.

Reply:
- I removed list endpoint now.

@coderbydesign 
Good point! Consumers will need to change how they will be call v2 endpoints than they did it before.
As those endpoints are new - I think that it will not break anything.